lib/mkl/interfaces.jl

```diff
@@ -31,3 +42,233 @@ end
 function LinearAlgebra.generic_trimatdiv!(C::oneMatrix{T}, uploc, isunitc, tfun::Function, A::oneSparseMatrixCSR{T}, B::oneMatrix{T}) where T <: BlasFloat
     sparse_trsm!(uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', 'N', isunitc, one(T), A, B, C)
 end
+
+# Handle Transpose and Adjoint wrappers for sparse matrices
+# Let the low-level wrappers handle the CSC->CSR conversion and flip_trans logic
+
+# Matrix-vector multiplication with transpose/adjoint
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Transpose{T, <:oneSparseMatrixCSR{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    tA_final = tA == 'N' ? 'T' : (tA == 'T' ? 'N' : 'C')
+    return sparse_gemv!(tA_final, _add.alpha, A.parent, B, _add.beta, C)
+end
+
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Adjoint{T, <:oneSparseMatrixCSR{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    if tA == 'T'
+        alpha = _add.alpha
+        beta = _add.beta
+        B .= conj.(B)
+        C .= conj.(C)
+        sparse_gemv!('N', conj(alpha), A.parent, B, conj(beta), C)
+        C .= conj.(C)
+        B .= conj.(B)
+    else
+        tA_final = tA == 'N' ? 'C' : 'N'
+        sparse_gemv!(tA_final, _add.alpha, A.parent, B, _add.beta, C)
+    end
+    return C
+end
+
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Transpose{T, <:oneSparseMatrixCSC{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    tA_final = tA == 'N' ? 'T' : (tA == 'T' ? 'N' : 'C')
+    return sparse_gemv!(tA_final, _add.alpha, A.parent, B, _add.beta, C)
+end
+
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Adjoint{T, <:oneSparseMatrixCSC{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    if tA == 'T'
+        alpha = _add.alpha
+        beta = _add.beta
+        B .= conj.(B)
+        C .= conj.(C)
+        sparse_gemv!('N', conj(alpha), A.parent, B, conj(beta), C)
+        C .= conj.(C)
+        B .= conj.(B)
+    else
+        tA_final = tA == 'N' ? 'C' : 'N'
+        sparse_gemv!(tA_final, _add.alpha, A.parent, B, _add.beta, C)
+    end
+    return C
+end
+
+function LinearAlgebra.generic_matvecmul!(C::oneVector{T}, tA::AbstractChar, A::Transpose{T, <:oneSparseMatrixCOO{T}}, B::oneVector{T}, _add::MulAddMul) where {T <: BlasFloat}
+    tA = tA in ('S', 's', 'H', 'h') ? 'N' : tA
+    tA_final = tA == 'N' ? 'T' : (tA == 'T' ? 'N' : 'C')
```

Comment:
Once again, special handle of `tA = 'C'` is needed (when the matrix is complex).